From 654d8841e67dba66125bfa4b8a6948bda9a187d2 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Fri, 15 Sep 2017 20:07:22 +0800
Subject: [PATCH 2536/5242] MLK-16461-2 ARM64: dts: fsl-imx8mq.dtsi: set
 property fsl,strobe-dll-delay-target

commit  0e8ac602372fd08b9dd965d82a0153d4f21baecf from
https://source.codeaurora.org/external/imx/linux-imx.git

On some i.MX8MQ-EVK A0/A1 board, we may meet eMMC read data CRC error in
HS400 mode. This is because the delay cell set for the strobe pad
is not correct. Before this patch, usdhc driver default set this delay
cell to value 7, but for some board, only value 3~6 can pass read
operation in HS400 mode, some board, the pass value is 2~10, and
some board, the pass value is 3~8. So the default value 7 can't
make sure all the board can pass eMMC read in HS400.

This patch set the value 5 to fsl,strobe-dll-delay-target, make sure
all the board can work stable in HS400 mode.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Reviewed-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 629f457..764de94 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -722,7 +722,8 @@
 		assigned-clocks = <&clk IMX8MQ_CLK_USDHC1_DIV>;
 		assigned-clock-rates = <400000000>;
 		fsl,tuning-start-tap = <20>;
-		fsl,tuning-step= <2>;
+		fsl,tuning-step = <2>;
+		fsl,strobe-dll-delay-target = <5>;
 		bus-width = <4>;
 		status = "disabled";
 	};
@@ -736,7 +737,7 @@
 			<&clk IMX8MQ_CLK_USDHC2_ROOT>;
 		clock-names = "ipg", "ahb", "per";
 		fsl,tuning-start-tap = <20>;
-		fsl,tuning-step= <2>;
+		fsl,tuning-step = <2>;
 		bus-width = <4>;
 		status = "disabled";
 	};
-- 
1.7.9.5

